File: src/interpreter/command/print.ts
import * as cal from 'calcium-js'

export class Print implements cal.Command {
  constructor(public readonly args: cal.Expression[]) {}

  execute(env: cal.Environment): cal.Behavior {
    console.log(this.args.map((a) => env.evaluate(a)).join(' '))
    return cal.Behavior.Forward
  }
}

File: lib/interpreter/pseudo/command/print.dart
import 'package:calcium_app/interpreter/behavior.dart';
import 'package:calcium_app/interpreter/command.dart';
import 'package:calcium_app/interpreter/expression.dart';
import 'package:calcium_app/interpreter/runtime_data.dart';

class Print implements Command {
  final List<Expression> args;

  Print(this.args);

  @override
  Behavior execute(RuntimeData data) {
    print(args.map((a) => a.evaluate(data).value).join(' '));
    return Behavior.forward;
  }
}
